src/otx/engine/hpo/hpo_api.py

```diff
@@ -133,8 +133,11 @@ def hpo_config(self) -> dict[str, Any]:
 
     @hpo_config.setter
     def hpo_config(self, hpo_config: HpoConfig | None) -> None:
-        train_dataset_size = len(self._engine.datamodule.subsets["train"])
-        val_dataset_size = len(self._engine.datamodule.subsets["val"])
+        train_dataset_size = len(self._engine.datamodule.train_dataloader())
+        if self._engine.datamodule.config.val_subset.subset_name in self._engine.datamodule.subsets:
```

Comment:
Raise an error if it cannot create `val_dataloader`

Reply:
As far as I know, anomaly task doesn't use validation dataset. It's for that.

Reply:
https://github.com/openvinotoolkit/training_extensions/blob/0b572ccdaef03f5541340b7eed62e37f03fc8a5a/src/otx/data/anomaly/anomaly.py#L52-L59

Reply:
Anyway, you must need an evaluation to do HPO. How do you HPO for anomaly tasks without a validation set?

Reply:
I found that anomaly models don't use both val_dataloader and test_dataloader. It seems that anomaly models use only train dataset for training.
HPO refers 'callback_monitor' as metric. So, `train_loss_epoch` is used for anomaly.

Reply:
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/7a52967e-19ac-42b9-9b88-b6d8fe6be50e)
I just examined it with my debugger and can find that the evaluation step exists and it comes from the test subset

Reply:
I removed them. Could you check?